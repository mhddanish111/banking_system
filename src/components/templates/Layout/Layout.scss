@import '../../../styles/boarder.scss';
.page-heading {
    border-bottom: $border-dark;
}