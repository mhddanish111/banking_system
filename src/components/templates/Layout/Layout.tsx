import React, { useEffect } from 'react';
import { isEmpty } from 'lodash';
import { useNavigate } from 'react-router-dom';
import Heading from '../../atoms/Heading';
import Header from '../../organisms/Header';
import Navigation from '../../organisms/Navigation';
import { getCookie } from '../../../utils';
import './Layout.scss';

type LayoutProps = {
  children: React.ReactNode;
  pageHeading?: String;
};
const Layout: React.FC<LayoutProps> = React.memo(
  ({ children, pageHeading }) => {
    const navigate = useNavigate();
    const token = getCookie();
    useEffect(() => {
      if (isEmpty(token)) {
        return navigate('/');
      }
    }, [token, navigate]);
    return (
      <>
        <Header />
        <div className="main-container">
          <Navigation />
          <main className="main">
            {pageHeading && (
              <Heading
                HeadingType="h1"
                className="page-heading"
                ariaLabel={pageHeading}
              >
                {pageHeading}
              </Heading>
            )}
            {children}
          </main>
        </div>
      </>
    );
  },
);

export default Layout;
