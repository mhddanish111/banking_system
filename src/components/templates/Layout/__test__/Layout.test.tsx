import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import Layout from '../Layout';

describe('<Layout />', () => {
  const props = {
    pageHeading: 'Test',
  };
  const mockedUsedNavigate = jest.fn();
  jest.mock('react-router-dom', () => ({
    ...(jest.requireActual('react-router-dom') as any),
    useNavigate: () => mockedUsedNavigate,
  }));
  const wrapped = shallow(
    <BrowserRouter>
      <Layout {...props}>
        <div>Test</div>
      </Layout>
    </BrowserRouter>,
  );
  it('should render the Layout Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
