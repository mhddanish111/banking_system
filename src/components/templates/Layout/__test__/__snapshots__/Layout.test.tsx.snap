// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render the Layout Component correctly 1`] = `
<Router
  location={
    Object {
      "hash": "",
      "key": "default",
      "pathname": "/",
      "search": "",
      "state": null,
    }
  }
  navigationType="POP"
  navigator={
    Object {
      "action": "POP",
      "createHref": [Function],
      "encodeLocation": [Function],
      "go": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "default",
        "pathname": "/",
        "search": "",
        "state": null,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Layout
    pageHeading="Test"
  >
    <div>
      Test
    </div>
  </Layout>
</Router>
`;
